src/transformation/visitors/class/index.ts

```diff
@@ -45,7 +45,14 @@ export function transformClassAsExpression(
     context: TransformationContext,
     isDefaultExport = false
 ): lua.Expression {
-    const className = isDefaultExport ? createDefaultExportIdentifier(expression) : lua.createAnonymousIdentifier();
+    let className: lua.Identifier;
```

Comment:
Have you tried removing that part? I don't really see why we need to infer local name twice

Reply:
It's tricky because of line 61

Reply:
Can we do something like this?
```ts
export function transformClassAsExpression(
    node: ts.ClassLikeDeclaration,
    context: TransformationContext
): lua.Expression {
    const { statements, className } = transformClassLikeDeclaration(node, context);
    return createImmediatelyInvokedFunctionExpression(statements, className, node);
}

export const transformClassDeclaration: FunctionVisitor<ts.ClassDeclaration> = (node, context) => {
	const { statements } = transformClassLikeDeclaration(node, context);
	return statements;
}
```